py_zipkin/zipkin.py

```diff
@@ -206,16 +206,17 @@ def start(self):
         # If zipkin_attrs are passed in or this span is doing its own sampling,
         # it will need to actually log spans at __exit__.
         self.perform_logging = self.zipkin_attrs or self.sample_rate is not None
+        report_root_timestamp = False
 
         if self.sample_rate is not None:
-            # This clause allows for sampling this service independently
-            # of the passed-in zipkin_attrs.
             if self.zipkin_attrs and not self.zipkin_attrs.is_sampled:
```

Comment:
Yeah, py-zipkin actually leaves the header extraction to whatever's calling it, e.g. pyramid-zipkin. It just lets you pass in arbitrary headers, which it calls zipkin_attrs, then automatically joins the upstream client span (related to comment above about the outermost `zipkin_span`), always logging ('sr', 'ss').

The brave model is more expressive and complete, but this gets the job done for most sane use cases.